Sources/MongoSwift/MongoError.swift

```diff
@@ -1,6 +1,128 @@
 import Foundation
 import mongoc
 
+/// An empty protocol for encapsulating all errors that this package can throw.
+public protocol MongoSwiftError: Error {}
+
+/// The possible errors corresponding to types of errors encountered in the MongoDB server.
+/// These errors may contain labels providing additional information on their origin.
+/// - SeeAlso: https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err
+public enum ServerError: MongoSwiftError, Equatable {
+    /// Thrown when commands experience errors on the server that prevent execution.
+    case commandError(code: Int, message: String, errorLabels: [String]?)
+
+    /// Thrown when errors occur on the server during commands that write not as part of a bulk write
```

Comment:
this wording is a little confusing... but off the top of my head I do not have a better way to put it... 🤔 maybe see how PHP/other drivers describe this case?
(and period at end of sentence please)

Reply:
PHP says:
>Base class for exceptions thrown by a failed write operation. The exception encapsulates a MongoDB\Driver\WriteResult object.

How does "Thrown when a single write command fails on the server." sound?

Reply:
sounds good to me.